dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/Filter.java

```diff
@@ -43,26 +43,26 @@
 @SPI
 public interface Filter {
 
+    default void beforeRequest(Invoker<?> invoker, Invocation invocation) {
+
+    }
+
+    default void afterRequest(Invoker<?> invoker, Invocation invocation) {
+
+    }
+
     /**
-     * do invoke filter.
-     * <p>
-     * <code>
-     * // before filter
-     * Result result = invoker.invoke(invocation);
-     * // after filter
-     * return result;
-     * </code>
-     *
-     * @param invoker    service
-     * @param invocation invocation.
-     * @return invoke result.
-     * @throws RpcException
-     * @see org.apache.dubbo.rpc.Invoker#invoke(Invocation)
+     * Does not need to override/implement this method.
      */
-    Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcException;
+    default Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcException {
+        return invoker.invoke(invocation);
+    }
+
+    default void onResponse(Result result, Invoker<?> invoker, Invocation invocation) {
 
-    default Result onResponse(Result result, Invoker<?> invoker, Invocation invocation) {
-        return result;
     }
 
+    default void onError(Throwable t, Invoker<?> invoker, Invocation invocation) {

```

Comment:
I am thinking of grouping `onResponse` `onError` into a `Listener` interface, they are  all callbacks and should not being defined directly in `Filter`.